### ISSUE TITLE
How do you train the model?

### ISSUE THREAD

**Issue by dreamhua82** [YingqianWang/iPASSR#2]
Hi, thanks for sharing.
I read your Stereo SR and LF SR code, but there are no validation code, so I want to know how do you find the best model(e.g. in iPASSRnet, the 80 epochs). Or, you just manually  use the test code every epoch to do validation? BTW, when do you share the LFDFNet code?  Is it possible to use the PAM module for LF SR ?
Again, thks a lot.

Waiting for your kindly reply,
Regards,

**Comment by YingqianWang** [comment]
Thanks for your attention and comments.

1) We did not use the validation set for both stereo image SR and LF image SR. During our training phase, we performed inference on the test set every epoch. Since the PSNR and SSIM were stable at the end of the training, we directly used the model at the 80th epoch as the final stereo image SR model and the model at the 50th epoch as the final LF image SR model.

2) We are recently busy in preparing the Chinese New Year, so the code of LF-DFnet may be released in the late Feburary. Sorry for the delay. It is worth noting that, we have been trying to modify our LF image SR recorces to make them easy to follow and implement. For example, we have cropped LF images into patches at the inference phase so that the test code can be implemented on GPUs. We hope that these modifications can facilitate the community and contribute to the development in LF image SR.

3) PAM has been successfully applied to LF reconstruction (also known as LF angular SR) in the work done by Wu et al. [c1]. From my point of view, there are two main challenges in applying PAM for LF image SR. The first challenge is how to build correspondence among multiple views since PAM is originally proposed to build correspondence between only two views. The second challenge is how to reduce the memory footprint to improve the computational efficiency since PAM will introduce a large GPU memory cost when applying to LF images. 

Welcome to pay continuous attention to our work.

[c1] Spatial-Angular Attention Network for Light Field Reconstruction, arXiv 2020.
